### PR TITLE
Added proper api to create context propagation

### PR DESCRIPTION
**Pull request by chp9-u**
Added the proper api to do context propagation according to [opentracing specs](https://opentracing.io/docs/overview/inject-extract/#the-basic-approach-inject-extract-and-carriers).  Fixed a problem where Process pid was set twice.

Requires:
https://github.com/team-at-cpan/Net-Async-OpenTracing/pull/3

### CODE REVIEW

File: examples/jaeger.pl [hunk 1]
```diff
@@ -77,6 +77,19 @@
 $log->infof('Create span via HTTP::Tiny integration');
 HTTP::Tiny->new->get('http://localhost');
 
+$log->infof('Create span via System integration');
+system('echo Created span via System integration');
```
Comment (chp9-u) [inline examples/jaeger.pl:81]: Added a test to use System Integration

File: examples/jaeger.pl [hunk 2]
```diff
@@ -77,6 +77,19 @@
...
+
+$log->infof('Inject span for Context Propagation');
+my $span_for_context = $tracer->span(operation_name => "span_context");
+my $payload = $tracer->inject($span_for_context);
+
+$log->infof('Extract span for Context Propagation');
+my $span_context = $tracer->extract($payload);
+my $span_with_context = $tracer->span(operation_name => "span_with_context", references => [$tracer->child_of($span_context)]);
+$span_with_context->finish;
+$span_for_context->finish;
```
Comment (chp9-u) [inline examples/jaeger.pl:91]: Added a test to simulate context propagation

File: lib/OpenTracing/Process.pm [hunk 3]
```diff
@@ -42,14 +42,6 @@ The process name. Freeform text string.
...
 sub name { shift->{name} //= "$0" }
 
-=head2 pid
-
-The process pid.
-
-=cut
-
-sub pid { shift->{pid} //= $$ }
-
```
Comment (chp9-u) [inline lib/OpenTracing/Process.pm:52]: Somehow pid was set twice in this file

File: lib/OpenTracing/Tracer.pm [hunk 4]
```diff
@@ -152,8 +195,9 @@ sub finish_span {
 
     @{ $self->{current_span} }  = grep { refaddr($_) != refaddr($span)} @{ $self->{current_span} };
 
-    return $span unless $self->is_enabled;
     push @{$self->{finished_spans} //= []}, $span;
```
Comment (chp9-u) [inline lib/OpenTracing/Tracer.pm:198]: finished_spans where not recorded in jaeger.pl, this fixed it

File: lib/OpenTracing/Tracer.pm [hunk 5]
```diff
@@ -212,20 +256,6 @@ sub span_for_future {
...
-
-    die "No context was provided" unless $args{context};
-
-    my $span_context = OpenTracing::SpanContext->new(span => $args{context});
-    $self->add_span(
-        my $span = $span_context->new_span($args{operation_name}, %args)
-    );
-    push @{ $self->{current_span} }, $span;
-    return OpenTracing::SpanProxy->new(span => $span);
-}
```
Comment (chp9-u) [inline lib/OpenTracing/Tracer.pm:227]: Replaced in favor of using:
```
$tracer->span(operation_name => "main", references => [$tracer->child_of($some_context)]);
```

**Comment by chp9-u** [comment]
Using 
```
docker run -p6831:6831/udp -p6832:6832/udp -p16686:16686 jaegertracing/all-in-one:latest --log-level=debug
```

Using a test Perl script, this results in:
localhost_16686_trace_e227751ca1be43f0756aecf197c84951

This also was tested with OpenTelemetry using agent configuration:
```
receivers:
  # https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/jaegerreceiver/README.md
  jaeger:
    protocols:
      thrift_binary: # 0.0.0.0:6832
      thrift_compact: # 0.0.0.0:6831

exporters:
  logging:
    loglevel: debug
  jaeger:
    endpoint: jaeger:14250
    insecure: true
    
service:
  pipelines:
    traces:
      receivers:
        - jaeger
      exporters:
        - logging
        - jaeger
```

And docker compose:
```
version: "2"
services:

  # Jaeger - https://hub.docker.com/r/jaegertracing/all-in-on
  jaeger:
    image: jaegertracing/all-in-one:latest
    ports:
      - "16686:16686" # HTTP, query, serve frontend

  # OpenTelemetry Agent - https://hub.docker.com/r/otel/opentelemetry-collector
  otel-agent:
    image: otel/opentelemetry-collector:latest
    volumes:
      - ./otel-agent-config.yaml:/etc/otel/config.yaml
    ports:
      - "6831:6831/udp"   # Jaeger - thrift_binary
      - "6832:6832/udp"   # Jaeger - thrift_binary
    depends_on:
      - jaeger
```

**Comment by tm604** [comment]
Thanks - confirmed Jæger is showing something sensible when running the examples, so I'll merge this and will be releasing a new version this week.

Any preferences on how you're credited in the documentation?

**Pull request merged**